tests/js/session-tests.js

```diff
@@ -177,6 +177,39 @@ module.exports = {
             });
     },
 
+    testRealmOpenWithDestructiveSchemaUpdate() {
+        if (!isNodeProcess) {
+            return;
+        }
+
+        const partition = Utils.genPartition();
+        const expectedObjectsCount = 3;
+
+        let user, config;
+        let credentials = Realm.Credentials.anonymous();
+        let app = new Realm.App(appConfig);
+        return runOutOfProcess(__dirname + '/download-api-helper.js', appConfig.id, appConfig.url, partition, REALM_MODULE_PATH)
+            .then(() => { return app.logIn(credentials) })
+            .then(u => {
+                user = u;
+                config = getSyncConfiguration(u, partition);
+                return Realm.open(config)
+            }).then(realm => {
+                realm.close();
+                // change the 'breed' property from 'string?' to 'string'
+                config.schema[0].properties.breed = "string";
+            }).then(() => {
+                return Realm.open(config);
+            }).then((realm) => {
+                // this should never happen (a better way of doing this?)
```

Comment:
See `TestCase.assertThrowsAsyncContaining` above. Perhaps that (and `assertThrowsAsync`) should be modified to check if the argument is already a promise and not call it, then you could just pass `Realm.open(...)` without wrapping in an extra lambda, but even without that change, I think it is an improvement. Also, IIUC, your version would still pass even if it failed on line 196, even though it is supposed to only be checking for a failure on line 202. You *could* solve that by putting the `.catch()` on line 202 directly on the `Realm.open()` call rather than at the top-level chain, but I think using the helper (and doing the await transform) is a better solution. (Technically that would also catch an immediate error in Realm.open() before it returns the promise, but I think that is OK)